src/main/java/io/quarkus/gizmo/MethodCreatorImpl.java

```diff
@@ -75,6 +78,22 @@ public AnnotatedElement getParameterAnnotations(int param) {
         return p;
     }
 
+    @Override
+    public void setParameterNames(String[] parameterNames) {
+        if (parameterNames == null) {
```

Comment:
Shouldn't it set `this.parameterNames` to `null`?

Reply:
That's a good idea. Removing what was already set may occasionally be useful.

Reply:
Done.